lib/app/modules/bulk_data/bulk_data_export_sequence.rb

```diff
@@ -0,0 +1,305 @@
+
+        reply = export_kick_off(endpoint)
+        @client.set_bearer_token(@instance.token)
+        assert_response_unauthorized reply
+      end
+
+      test 'Server shall return "202 Accepted" and "Content-location" for $export operation with parameters' do
+        metadata do
+          id '02'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/export/index.html#query-parameters'
+          desc %(
+          )
+        end
+
+        check_export_kick_off(search_params: { '_type' => endpoint })
+      end
+
+      test 'Server shall return "202 Accepted" and "Content-location" for $export operation' do
+        metadata do
+          id '03'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/export/index.html#bulk-data-kick-off-request'
+          description %(
+          )
+        end
+
+        skip 'Skip testing $export without parameters' if @server_supports_type_parameter && !run_all_kick_off_tests
+
+        check_export_kick_off
+      end
+
+      test 'Server shall reject for $export operation with invalid Accept header' do
+        metadata do
+          id '04'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/export/index.html#headers'
+          description %(
+          )
+        end
+
+        check_export_kick_off_fail_invalid_accept
+      end
+
+      test 'Server shall reject for $export operation with invalid Prefer header' do
+        metadata do
+          id '05'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/export/index.html#headers'
+          description %(
+          )
+        end
+
+        check_export_kick_off_fail_invalid_prefer
+      end
+
+      test 'Server shall return "202 Accepted" or "200 OK"' do
+        metadata do
+          id '06'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/export/index.html#bulk-data-status-request'
+          description %(
+          )
+        end
+
+        check_export_status
+      end
+
+      test 'Server shall return file in ndjson format' do
+        metadata do
+          id '07'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/export/index.html#file-request'
+          description %(
+          )
+        end
+
+        check_file_request
+      end
+
+      test 'Server should return "202 Accepted" for delete export content' do
+        metadata do
+          id '08'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/export/index.html#bulk-data-delete-request'
+          desc %(
+          )
+          optional
+        end
+
+        check_delete_request(@content_location)
+      end
+
+      test 'Server shall return "202 Accepted" for cancel export request' do
+        metadata do
+          id '09'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/export/index.html#bulk-data-delete-request'
+          desc %(
+          )
+          optional
+        end
+
+        check_cancel_request
+      end
+
+      private
+
+      def export_kick_off(endpoint = nil,
```

Comment:
It's confusing to me that this default value is `nil`, but `endpoint` is set to `''`. If this argument is always being passed in, i think we should get rid of the default value.